src/app/api/cronjob/route.ts

```diff
@@ -0,0 +1,96 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import {
+  addDays,
+  eachDayOfInterval,
+  format,
+  isWithinInterval,
+  subDays,
+} from "date-fns";
+import { and, eq, gte, lte } from "drizzle-orm";
+
+import { db } from "@/lib/db";
+import {
+  gameSessions,
+  gameSessionSchedules,
+  semesters,
+  weekdayEnum,
+} from "@/lib/db/schema";
+import { insertGameSessionSchema } from "@/lib/validators";
+
+export async function POST(req: NextRequest) {
+  try {
+    const now = new Date();
+    const oneWeekFromNow = addDays(now, 7);
+
+    const activeSemester = await db.query.semesters.findFirst({
+      where: and(
+        lte(semesters.startDate, format(now, "yyyy-mm-dd")),
+        gte(semesters.endDate, format(now, "yyyy-mm-dd")),
+      ),
+    });
+
+    if (!activeSemester) {
+      return NextResponse.json("No active semester found.", { status: 400 });
+    }
+
+    if (
+      activeSemester.breakStart &&
+      activeSemester.breakEnd &&
```

Comment:
is this for checking whether they exist? `breakStart` and `breakEnd` are nonNull fields so you shouldn't need this

Reply:
Yeah whoops, I missed that. Removed.